app/controllers/clients_controller.rb

```diff
@@ -49,11 +51,13 @@ def set_client
   end
 
   def client_params
-    params.require(:client).permit(:id, :first_name, :last_name, :date_of_birth, :nationality,
-      :permit, :gender, :street, :zip, :city, :phone, :email, :goals, :education, :hobbies,
+    params.require(:client).permit(:date_of_birth, :nationality,
+      :permit, :gender, :goals, :education, :hobbies,
       :interests, :state, :comments, :competent_authority, :involved_authority, :user_id,
       language_skills_attributes: language_skills_attributes,
       relatives_attributes: relatives_attributes,
-      schedules_attributes: schedules_attributes)
+      schedules_attributes: schedules_attributes,
+      contact_attributes: contact_attributes
+      )
```

Comment:
in the parameter returning methods, do:

```ruby
def contact_attributes
  { contact_attributes: [:bogus, :example, :more, :attr] }
end
```

instead of:
```ruby
def contact_attributes
  [:bogus, :example, :more, :attr]
end
```

then you can shorten this:

```ruby
language_skills_attributes: language_skills_attributes,
relatives_attributes: relatives_attributes,
schedules_attributes: schedules_attributes,
contact_attributes: contact_attributes
```
to:
```ruby
relatives_attributes, language_skills_attributes, schedules_attributes, contact_attributes
```

instead of the: